simulator/func_sim/alu.h

```diff
@@ -215,6 +215,18 @@ struct ALU
     // Bit permutation
     template<typename I> static void grev( I* instr) { instr->v_dst[0] = gen_reverse( instr->v_src[0], shamt_v_src2<typename I::RegisterUInt>( instr)); }
 
+    template<typename I> static
+    void riscv_shfl( I* instr)
+    {
+	auto dst_value = instr->v_src[0];
```

Comment:
We do not use tabs. See https://github.com/MIPT-ILab/mipt-mips/wiki/Code-style-guidelines#indentation